xdsl/ir.py

```diff
@@ -437,7 +437,7 @@ def print_parameter(self, printer: Printer) -> None:
 class ParametrizedAttribute(Attribute):
     """An attribute parametrized by other attributes."""
 
-    parameters: list[Attribute] = field(default_factory=list)
+    parameters: Sequence[Attribute] = field(default_factory=list)
```

Comment:
This won't work, we want the class to know the exect type of its stored property

Reply:
right i will fix this